### PR TITLE
Add after limit joint command publisher to march_hardware_interface

### PR DESCRIPTION
**Pull request by JitskedeVries**
The same pullrequest as https://github.com/project-march/hardware-interface/pull/116, but no unnecessary changes. 

- Still somehow some style things were adjusted. Also 

- This pullrequest publishes the commands after the limits and which commands actually are sent towards the IMC.

- A small refactor of the write method with the new joint.getname function

### CODE REVIEW

File: march_hardware_interface/src/march_hardware_interface.cpp [hunk 1]
```diff
@@ -372,6 +380,31 @@ void MarchHardwareInterface::updatePowerNet()
   }
 }
 
+void MarchHardwareInterface::updateAfterLimitJointCommand()
+{
+  return;
```
Comment (martijnvandermarel) [inline march_hardware_interface/src/march_hardware_interface.cpp]: ??? Return shouldn't be here

**Review by martijnvandermarel — CHANGES_REQUESTED** [review]
Maybe change back the style changes?
Functionality looks good, except for the weird return right at the start of the method?

File: march_hardware_interface/src/march_hardware_interface.cpp [hunk 2]
```diff
@@ -372,6 +380,31 @@ void MarchHardwareInterface::updatePowerNet()
...
+  }
+
+  // Clear msg of AfterLimitJointCommand
+  after_limit_joint_command_pub_->msg_.name.clear();
+  after_limit_joint_command_pub_->msg_.position_command.clear();
+  after_limit_joint_command_pub_->msg_.effort_command.clear();
+
+  for (int i = 0; i < num_joints_; i++)
+  {
+    march4cpp::Joint singleJoint = marchRobot.getJoint(joint_names_[i]);
```
Comment (Ishadijcks) [inline march_hardware_interface/src/march_hardware_interface.cpp]: I think `joint` is sufficient, more consistent with other code and I don't understand what `single` adds here.

File: march_hardware_interface/src/march_hardware_interface.cpp (hunk 1 above)
Comment (Ishadijcks) [inline march_hardware_interface/src/march_hardware_interface.cpp]: Huh?

File: march_hardware_interface/src/march_hardware_interface.cpp [hunk 3]
```diff
@@ -41,6 +41,10 @@ void MarchHardwareInterface::init()
   imc_state_pub_ = RtPublisherPtr(
       new realtime_tools::RealtimePublisher<march_shared_resources::ImcErrorState>(this->nh_, "/march/imc_states/", 4));
 
+  after_limit_joint_command_pub_ = RtPublisherAfterLimitJointCommandPtr(
+      new realtime_tools::RealtimePublisher<march_shared_resources::AfterLimitJointCommand>(
+          this->nh_, "/march/controller/after_limit_joint_command/", 4));
```
Comment (Ishadijcks) [inline march_hardware_interface/src/march_hardware_interface.cpp:46]: Better!

File: march_hardware_interface/include/march_hardware_interface/march_hardware_interface.h [hunk 4]
```diff
@@ -25,17 +27,18 @@ static const double VELOCITY_STEP_FACTOR = 10;
...
+  MarchHardwareInterface(ros::NodeHandle& nh, AllowedRobot robotName);
   ~MarchHardwareInterface();
 
   /**
    * @brief Initialize the HardwareInterface by registering position interfaces
    * for each joint.
    */
   void init();
-  void update(const ros::TimerEvent &e);
+  void update(const ros::TimerEvent& e);
```
Comment (Ishadijcks) [inline march_hardware_interface/include/march_hardware_interface/march_hardware_interface.h]: Please revert unneeded style changes.

**Review by Ishadijcks — CHANGES_REQUESTED** [review]

**Review by martijnvandermarel — APPROVED** [review]

**Review by Ishadijcks — APPROVED** [review]

**Pull request merged**